### PR TITLE
Adding notes about particle deposit fields.

### LINKED ISSUE #2218: Update docs for deposited particle fields
The "Deposited Particle Fields" and following sections in `doc/source/analyzing/fields.rst` need to be updated to incorporate changes in yt-4.0. I've left a `FIXME` there. This issue is to make sure it gets done before yt-4.0.

Related issues:
- #2205
- #2192

### PR DESCRIPTION
**Pull request by matthewturk**
This adds -- and removes! -- documentation about particle deposit fields.

Fixes #2218.

### CODE REVIEW

**Comment by jzuhone** [comment]
Would this be the place to mention `ParticleProjectionPlot` or should that be somewhere else? We already have docs for it but it would be good to link it somewhere.

**Comment by matthewturk** [comment]
Sure -- would you mind suggesting a change?

**Comment by jzuhone** [comment]
Yes, I'll do that.

**Comment by neutrinoceros** [comment]
> Doc build might be failing because I'm referencing a section that won't exist until #3293 goes in.

Let's try now !
@yt-fido test this please

**Review by cphyc — APPROVED** [review]

File: doc/source/analyzing/fields.rst [hunk 1]
```diff
@@ -704,8 +704,16 @@ defined on each dataset to depose any particle field onto the mesh like so:
...
-   print(ds.r["deposit", "all_nn_particle_velocity_x"])
+   print("The velocity of the particles are (stored in %s)" % (fname,))
+   print(ds.r[fname])
+
+.. note::
+
+   In this example, we are using the returned field name as our input.  You
+   *could* also access it directly, but it might take a slightly different form
+   than you expect -- in this particular case, the field name will be
+   ``("deposit", ""all_nn_velocity_x")``, which has removed the prefix
```
[the remark this answers is not in the document]
Comment (neutrinoceros) [reply doc/source/analyzing/fields.rst:715]: My approval is only conditional on this typo being fixed

**Pull request merged**